pkg/provider/aws/auth/auth.go

```diff
@@ -70,6 +66,15 @@ const (
 	errMissingAKID                             = "missing AccessKeyID"
 )
 
+func init() {
+	fs := pflag.NewFlagSet("aws-auth", pflag.ExitOnError)
+	fs.BoolVar(&enableSessionCache, "experimental-enable-aws-session-cache", false, "Enable experimental AWS session cache. External secret will reuse the AWS session without creating a new one on each request.")
+	feature.Register(feature.Feature{
+		Flags: fs,
+	})
+	sessionCache = cache.Must[*session.Session](1024, nil)
```

Comment:
I think we've already discussed this, but we don't need to do a cleanup here because AWS does the cleanup for us after a fixed amount of time, right? 😅

Reply:
revisiting this after a long time i had to dig for this a bit: 
* yes, we don't need to clean that up. session duration is 1-12h in AWS
* in case the AWS session expires the `*session.Session` will take care of renewing the credentials in the background. It is not needed with static credentials; It is done when using creds via Kubernetes service account. The `jwtProvider` generates a new Kubernetes SA token: this process is automatic and happening in the background

Speaking of expiration and cleanup: There is a [LRU with ttl PR](https://github.com/hashicorp/golang-lru/pull/116) in flight which looks very promising. It doesn't fit 100% our needs but this looks like a very good starting point. We can consider building on top of a LRU with ttl which provides a way to define eviction/expiration/refresh callbacks so we can take care of renewing them in the background. But that's something for the future.